Userland/Games/ColorLines/ColorLines.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "AK/NonnullRefPtr.h"
+#include "AK/RefPtr.h"
+#include "AK/Vector.h"
+#include "MarblePath.h"
+#include <LibGUI/Frame.h>
+#include <LibGfx/Font/BitmapFont.h>
+#include <LibGfx/Forward.h>
+
+class MarbleBoard;
+
+class ColorLines : public GUI::Frame {
+    C_OBJECT(ColorLines);
+
+public:
+    virtual ~ColorLines() override = default;
+
+    void reset();
+
+private:
+    enum class GameState {
+        Idle = 0,          // No marble is selected, waiting for marble selection
+        StartingGame,      // Game is starting
+        GeneratingMarbles, // Three new marbles are being generated
+        MarbleSelected,    // Marble is selected, waiting for the target cell selection
+        MarbleMoving,      // Selected marble is moving to the target cell
+        MarblesRemoving,   // Selected marble has completed the move and some marbles are being removed from the board
+        CheckingMarbles,   // Checking whether marbles on the board form lines of 5 or more marbles
+        GameOver           // Game is over
+    };
+
+    ColorLines(StringView app_name);
+
+    virtual void paint_event(GUI::PaintEvent&) override;
+    virtual void mousedown_event(GUI::MouseEvent&) override;
+    virtual void timer_event(Core::TimerEvent&) override;
+
+    void set_game_state(GameState state);
+    void restart_timer(int milliseconds);
+
+    using Point = Gfx::IntPoint;
+    using BitmapArray = Vector<NonnullRefPtr<Gfx::Bitmap>>;
+
+    StringView const m_app_name;
+    GameState m_game_state { GameState::Idle };
+    NonnullOwnPtr<MarbleBoard> m_board;
+    BitmapArray const m_marble_bitmaps;
```

Comment:
If we know how many marble colors we have at compile-time, we should probably use an `Array` of `RefPtr`s, like:

`Array<RefPtr<Gfx::Bitmap>, Marble::number_of_colors> m_marble_bitmaps {};`

Reply:
If you'd need to pass the array around places, you can call `span()` on `Array` to get a `Span<T const> const>`, which is non-owning view.